src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactorBackupCodesActions.tsx

```diff
@@ -0,0 +1,61 @@
+import { Button, BorderBoxProps, Flex } from "@artsy/palette"
+import React, { useEffect, useState } from "react"
+
+interface SettingsEditSettingsTwoFactorBackupCodesActionsProps
+  extends BorderBoxProps {
+  backupSecondFactors: string[]
+}
+
+export const SettingsEditSettingsTwoFactorBackupCodesActions: React.FC<SettingsEditSettingsTwoFactorBackupCodesActionsProps> = props => {
+  const { backupSecondFactors } = props
+
+  const [supportsClipboard, setSupportsClipboard] = useState(false)
+
+  useEffect(() => {
+    // Only render the copy button if browser supports the Clipboard API
+    // https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API
+    if ("clipboard" in navigator) setSupportsClipboard(true)
```

Comment:
This is a good check, but I _think_ all browsers now support this. In any case, this makes things a bit safer.

Reply:
For sure. Fortunately, all major browsers do support it now, but Edge and Safari support are more recent than I would have thought (2020).